src/jit/codegencommon.cpp

```diff
@@ -7557,6 +7557,7 @@ void CodeGen::genPrologPadForReJit()
 regMaskTP           CodeGen::genPInvokeMethodProlog(regMaskTP initRegs)
 {
     assert(compiler->compGeneratingProlog);
+    noway_assert((compiler->opts.eeFlags & CORJIT_FLG_PINVOKE_USE_HELPERS) == 0);
```

Comment:
Yes, I thought about this, but it didn't seem worth it at the time. I'll go ahead and make the suggested change.